src/components/Button/Button.scss

```diff
@@ -392,6 +392,11 @@ $stacking-order: (
       opacity: 0.4;
     }
 
+    // Prevents the loading spinner from overlapping the button’s text, while retaining its width.
+    &.loading .Text {
+      opacity: 0;
```

Comment:
^This is a [known issue](https://github.com/Shopify/polaris-react/issues/1212) 💯 

```suggestion
      visibility: hidden;
```